src/components/pages/datasets/cards/datasets.tsx

```diff
@@ -1,21 +1,15 @@
 import type {FunctionComponent} from 'react'
 import {Card} from '@/components/lib'
+import {IDataset} from '@/types/datasets'
 
-interface IDatasetCard {
-  title: string
-  description: string
-  tags: Array<string>
-  tensors: number
-}
-
-const DatasetCard: FunctionComponent<IDatasetCard> = ({title, description, tags, tensors}) => (
+const DatasetCard: FunctionComponent<IDataset> = ({name, description, tags, tensors}) => (
   <Card className="space-y-2 cursor-pointer">
-    <h2 className="text-xl text-gray-800 text-semibold">{title}</h2>
+    <h2 className="text-xl text-gray-800 text-semibold">{name}</h2>
     <p className="max-w-lg font-light text-gray-400 xl:max-w-5xl line-clamp-2">{description}</p>
     <div className="flex flex-col justify-between text-sm text-gray-600 md:flex-row">
       <small>{tags.map(tag => `#${tag} `)}</small>
       <small>
-        {tensors} tensor{tensors !== 1 && 's'}
+        {Object.keys(tensors).length} tensor{Object.keys(tensors).length !== 1 && 's'}
```

Comment:
Well, the api response for a dataset gives the tensors in form of a dictionary. We can change this to be returned in form of an array so that we can call array.length. Otherwise, we can add a key to the response json as the count of tensors. What are your thoughts?

Reply:
I thought of calling `Object.keys(tensors)` once in the main page and 
passing `tensors` as a number to the card.